lib/cn/route.h

```diff
@@ -141,11 +141,18 @@ route_node_key_modify(
     uint               edge_klen);
 
 static HSE_ALWAYS_INLINE int
-route_node_keycmp(const struct route_node *node, const void *key, uint klen)
+route_node_keycmp(const void *key, uint klen, const struct route_node *node)
 {
-    return keycmp(node->rtn_keybufp, node->rtn_keylen, key, klen);
+    return keycmp(key, klen, node->rtn_keybufp, node->rtn_keylen);
 }
 
+static HSE_ALWAYS_INLINE int
+route_node_keycmp_prefix(const void *pfx, uint pfxlen, const struct route_node *node)
+{
+    return keycmp_prefix(pfx, pfxlen, node->rtn_keybufp, node->rtn_keylen);
+}
```

Comment:
route_node * is the first argument to most of the route_node_*() functions, why did you move it?

Reply:
Yeah we do, but I felt it would be more natural to compare the key under consideration against the edge key. Especially in the case of the prefix key compare where the prefix is compared against the edge key's prefix.

Reply:
Not following, why not keep the route_node API consistent?

Reply:
Look at `keycmp_prefix(A, B)`. This compares A with B's prefix. Calling `route_node_keycmp_prefix(node, A)` may cause confusion if we're comparing A with the edge key's prefix or the edge key with A's prefix. So I thought it'd be consistent with `keycmp_prefix()` to have (A, node) as the arguments.

The `route_node_keycmp()` API could stay as it was, but I changed it so it's consistent with `route_node_keycmp_prefix()`.

Tbh, I don't feel too strongly about it. If it doesn't make sense to people, I can use the (node, A) order or arguments in both the calls.

Reply:
Gotcha..  Something weird about this API being part of route-map, can't put my finger on it...  :-)

Reply:
One option could be to have a non-copying version of `route_node_keycpy()` to just get a pointer to the edge key and its length. Then the caller can do `keycmp()` or `keycmp_prefix()` and we'd avoid incurring any unnecessary key copy.

But the onus is on the caller to make sure the edge key pointer is not used outside the tree lock section.